tests/integration/test_bundle.py

```diff
@@ -0,0 +1,46 @@
+import subprocess
+
+import pytest
+import requests
+from pytest_operator.plugin import OpsTest
+from tenacity import retry, stop_after_delay, wait_fixed
+
+BUNDLE_PATH = "./releases/latest/edge/mlflow/bundle.yaml"
+MLFLOW_APP_NAME = "mlflow-server"
+
+
+@pytest.fixture
+def forward_connections():
+    mlflow_process = subprocess.Popen(
+        ["kubectl", "-n", "kubeflow", "port-forward", "pod/mlflow-server-0", "5002:5000"]
+    )
+
+    exporter_process = subprocess.Popen(
+        ["kubectl", "-n", "kubeflow", "port-forward", "pod/mlflow-server-0", "8002:8000"]
+    )
+    yield
+    mlflow_process.terminate()
+    exporter_process.terminate()
+
+
+class TestCharm:
+    @pytest.mark.abort_on_fail
+    async def test_deploy_bundle_works(self, ops_test: OpsTest):
+        subprocess.Popen(["juju", "deploy", f"{BUNDLE_PATH}", "--trust"])
+        ops_test.deploy_bundle(BUNDLE_PATH)
+        await ops_test.model.wait_for_idle(
+            apps=[MLFLOW_APP_NAME],
+            status="active",
+            raise_on_blocked=False,
+            raise_on_error=False,
+            timeout=1500,
+        )
+
+    @retry(stop=stop_after_delay(5), wait=wait_fixed(1))
+    @pytest.mark.abort_on_fail
+    async def test_mlflow_connetion(self, forward_connections, ops_test: OpsTest):
+        mlflow_response = requests.get("http://localhost:5002")
```

Comment:
I 'm not sure I get why we have to `kubectl port-forward` here.

Reply:
Yeah, this actually is the way we did it also while developing WAs if we needed a component and didn't want to build it locally. It looks like yesterday I was a bit tired to understand what was happening here :P . Anyway, the only thought I have about this is if this is a proper way to test the connection to MLFlow, since we wouldn't `port-forward` anything during a normal deployment. Wouldn't it make more sense to try and hit its public URL?

Reply:
juju public url is juju think I wanted to test it more kubernetes like ... with port forward I can be 100% sure I can talk to it from cluster

Reply:
Okk, commenting just to add the context I missed in order to understand the situation here. At the time of writing, when deploying MLFlow standalone (without Kubeflow), there is no Istio integration or LoadBalancer support to expose MLFlow.